src/appsignal/cli/diagnose.py

```diff
@@ -0,0 +1,388 @@
+        params = urllib.parse.urlencode(
+            {
+                "api_key": self.config.option("push_api_key"),
+                "name": self.config.option("name"),
+                "environment": self.config.option("environment"),
+                "hostname": self.config.option("hostname") or "",
+            }
+        )
+
+        endpoint = self.config.option("diagnose_endpoint")
+        url = f"{endpoint}?{params}"
+
+        response = requests.post(url, json={"diagnose": self.report})
+
+        status = response.status_code
+        if status == 200:
+            token = response.json()["token"]
+            print()
+            print(f"  Your support token: {token}")
+            print(f"  View this report:   https://appsignal.com/diagnose/{token}")
+        else:
+            print(
+                "  Error: Something went wrong while submitting the report to AppSignal."
+            )
+            print(f"  Response code: {status}")
+            print(f"  Response body: {response.text}")
+
+    def _paths_data(self) -> dict:
+        working_dir = os.getcwd() or ""
+        log_file_path = self.config.log_file_path() or ""
+        log_path = self.config.option("log_path") or os.path.dirname(log_file_path)
+
+        try:
+            with open(log_file_path) as log_file:
+                lines = log_file.readlines()
+                log_file_contents = lines[-10:]
+                log_file_contents = [line.strip() for line in log_file_contents]
+        except FileNotFoundError:
+            log_file_contents = [""]
+
+        return {
+            "appsignal.log": {
+                "content": log_file_contents,
+                "exists": self._file_exists(log_file_path),
+                "mode": self._file_stat(log_file_path).get("mode"),
+                "ownership": self._file_stat(log_file_path).get("ownership"),
+                "path": log_file_path,
+                "type": self._file_type(log_file_path),
+                "writable": self._file_is_writable(log_file_path),
+            },
+            "log_dir_path": {
+                "exists": self._file_exists(log_path),
+                "mode": self._file_stat(log_path).get("mode"),
+                "ownership": self._file_stat(log_path).get("ownership"),
+                "path": log_path,
+                "type": self._file_type(log_path),
+                "writable": self._file_is_writable(log_path),
+            },
+            "working_dir": {
+                "exists": self._file_exists(working_dir),
+                "mode": self._file_stat(working_dir).get("mode"),
+                "ownership": self._file_stat(working_dir).get("ownership"),
+                "path": working_dir,
+                "type": self._file_type(working_dir),
+                "writable": self._file_is_writable(working_dir),
+            },
+        }
+
+    def _file_exists(self, path: str) -> bool:
+        return os.path.isfile(path) or os.path.isdir(path)
```

Comment:
Would `os.path.exists` also work here? Like is used in the `_file_type` method.

Reply:
Yup it should. This is the result of me starting to use the `Path` module and then switching its usage to `os` 🙃